src/auth.ts

```diff
@@ -1,15 +1,84 @@
 import * as vscode from "vscode";
 
-export const getApiKeyCredentials = () => {
-  const configurationWorkspace = vscode.workspace.getConfiguration();
-  const env0ApiKey: string | undefined =
-    configurationWorkspace.get("env0.apiKeyId");
-  const env0SecretKey: string | undefined =
-    configurationWorkspace.get("env0.secretKey");
+const env0KeyIdKey = "env0.keyId";
+const env0SecretKey = "env0.secret";
 
-  if (!env0ApiKey || !env0SecretKey) {
-    throw new Error("Could not read env0 api key values");
+export class AuthService {
+  public onAuth?: () => any = undefined;
+  constructor(private readonly context: vscode.ExtensionContext) {}
+  public registerLoginCommand() {
+    const disposable = vscode.commands.registerCommand(
+      "env0.login",
+      async () => {
+        const keyId = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: "API Key ID",
+          prompt: "Enter your API Key ID.",
+          password: true,
+          validateInput: (token) => {
```

Comment:
can we use some stricter rule? I'm sure we can get length or even a regex for what a valid key id and secret are allowed to look like

Reply:
10d886b7d760b16446a4a20c9a3c11bd38cd31b6